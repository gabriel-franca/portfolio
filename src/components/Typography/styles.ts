import styled, { css } from 'styled-components'

import { TypographyProps } from '.'

export const Wrapper = styled.span<TypographyProps>`
  ${({ theme, variant }) => css`
    font-size: ${theme.typography.fonts[variant!].fontSize};
    font-weight: ${theme.typography.fonts[variant!].fontWeight};
  `}
`
