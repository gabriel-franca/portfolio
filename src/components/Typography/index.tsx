import theme from '~/styles/theme'

import * as S from './styles'

export type TypographyProps = {
  variant?: keyof typeof theme.typography.fonts
  children?: string
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span'
}

const Typography = ({
  children,
  variant = 'body1',
  as = 'span'
}: TypographyProps) => (
  <S.Wrapper as={as} variant={variant}>
    {children}
  </S.Wrapper>
)

export default Typography
