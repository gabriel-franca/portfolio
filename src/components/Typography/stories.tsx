import { Meta, Story } from '@storybook/react/types-6-0'

import Typography, { TypographyProps } from '.'

export default {
  title: 'Typography',
  component: Typography,
  argTypes: {
    children: {
      type: 'string'
    }
  }
} as Meta

export const Default: Story<TypographyProps> = (args) => (
  <Typography {...args} />
)

Default.args = {
  children: 'Typography',
  variant: 'headline1'
}
