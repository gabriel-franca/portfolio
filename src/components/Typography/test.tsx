import { screen } from '@testing-library/react'
import { renderWithTheme } from '~/utils/tests/helpers'

import Typography from '.'

describe('<Typography />', () => {
  it('should render body1 as default', () => {
    const { container } = renderWithTheme(<Typography>typography</Typography>)

    expect(screen.getByText(/typography/i)).toHaveStyle({
      'font-size': '1.6rem',
      'font-weight': 'normal'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render headline1', () => {
    const { container } = renderWithTheme(
      <Typography variant="headline1">typography</Typography>
    )

    expect(screen.getByText(/typography/i)).toHaveStyle({
      'font-size': '4.8rem',
      'font-weight': 'normal'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render headline2', () => {
    const { container } = renderWithTheme(
      <Typography variant="headline2">typography</Typography>
    )

    expect(screen.getByText(/typography/i)).toHaveStyle({
      'font-size': '3.4rem',
      'font-weight': 'normal'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render headline3', () => {
    const { container } = renderWithTheme(
      <Typography variant="headline3">typography</Typography>
    )

    expect(screen.getByText(/typography/i)).toHaveStyle({
      'font-size': '2.4rem',
      'font-weight': 'normal'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render body2', () => {
    const { container } = renderWithTheme(
      <Typography variant="body2">typography</Typography>
    )

    expect(screen.getByText(/typography/i)).toHaveStyle({
      'font-size': '1.4rem',
      'font-weight': 'normal'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render caption', () => {
    const { container } = renderWithTheme(
      <Typography variant="caption">typography</Typography>
    )

    expect(screen.getByText(/typography/i)).toHaveStyle({
      'font-size': '1.2rem',
      'font-weight': 'normal'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render overline', () => {
    const { container } = renderWithTheme(
      <Typography variant="overline">typography</Typography>
    )

    expect(screen.getByText(/typography/i)).toHaveStyle({
      'font-size': '1rem',
      'font-weight': 'normal'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render span as default', () => {
    const { container } = renderWithTheme(<Typography>typography</Typography>)

    expect(container.querySelector('span'))
  })

  it('should render paragraph', () => {
    const { container } = renderWithTheme(
      <article>
        <Typography as="p">typography</Typography>
      </article>
    )

    // expect(container.querySelector('p')).toBeInTheDocument()
    expect(screen.getByRole('paragraph')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
