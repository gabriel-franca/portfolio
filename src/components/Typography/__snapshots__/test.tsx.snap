// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should render body1 as default 1`] = `
.c0 {
  font-size: 1.6rem;
  font-weight: normal;
}

<span
  class="c0"
>
  typography
</span>
`;

exports[`<Typography /> should render body2 1`] = `
.c0 {
  font-size: 1.4rem;
  font-weight: normal;
}

<span
  class="c0"
>
  typography
</span>
`;

exports[`<Typography /> should render caption 1`] = `
.c0 {
  font-size: 1.2rem;
  font-weight: normal;
}

<span
  class="c0"
>
  typography
</span>
`;

exports[`<Typography /> should render headline1 1`] = `
.c0 {
  font-size: 4.8rem;
  font-weight: normal;
}

<span
  class="c0"
>
  typography
</span>
`;

exports[`<Typography /> should render headline2 1`] = `
.c0 {
  font-size: 3.4rem;
  font-weight: normal;
}

<span
  class="c0"
>
  typography
</span>
`;

exports[`<Typography /> should render headline3 1`] = `
.c0 {
  font-size: 2.4rem;
  font-weight: normal;
}

<span
  class="c0"
>
  typography
</span>
`;

exports[`<Typography /> should render overline 1`] = `
.c0 {
  font-size: 1rem;
  font-weight: normal;
}

<span
  class="c0"
>
  typography
</span>
`;

exports[`<Typography /> should render p 1`] = `
.c0 {
  font-size: 1.6rem;
  font-weight: normal;
}

<p
  class="c0"
>
  typography
</p>
`;

exports[`<Typography /> should render p as default 1`] = `
.c0 {
  font-size: 1.6rem;
  font-weight: normal;
}

<p
  class="c0"
>
  typography
</p>
`;

exports[`<Typography /> should render paragraph 1`] = `
.c0 {
  font-size: 1.6rem;
  font-weight: normal;
}

<p
  class="c0"
>
  typography
</p>
`;

exports[`<Typography /> should render span as default 1`] = `
.c0 {
  font-size: 1.6rem;
  font-weight: normal;
}

<p
  class="c0"
>
  typography
</p>
`;
